target/linux/ath79/image/generic.mk

```diff
@@ -85,6 +85,15 @@ define Device/8dev_carambola2
 endef
 TARGET_DEVICES += 8dev_carambola2
 
+define Device/alfa_ap121f
+  ATH_SOC := ar9331
+  DEVICE_TITLE := ALFA Network AP121F
+  DEVICE_PACKAGES := -swconfig
+  IMAGE_SIZE := 16064k
+  SUPPORTED_DEVICES += ap121f
```

Comment:
The ath79-based firmware can be sysupgraded from a previous ar71xx without the -F option-